### PR TITLE
feat: projections backoffice

### PR DESCRIPTION
**Pull request by jvandaal**

### CODE REVIEW

File: src/ParcelRegistry.Projections.BackOffice/appsettings.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+﻿{
+  "ConnectionStrings": {
+    "Events": "Server=(localdb)\\mssqllocaldb;Database=EFProviders.InMemory.ParcelRegistry;Trusted_Connection=True;TrustServerCertificate=True;",
+    "BackOffice": "Server=(localdb)\\mssqllocaldb;Database=EFProviders.InMemory.ParcelRegistry;Trusted_Connection=True;TrustServerCertificate=True;",
+    "BackOfficeProjections": "Server=(localdb)\\mssqllocaldb;Database=EFProviders.InMemory.ParcelRegistry;Trusted_Connection=True;TrustServerCertificate=True;",
+    "BackOfficeProjectionsAdmin": "Server=(localdb)\\mssqllocaldb;Database=EFProviders.InMemory.ParcelRegistry;Trusted_Connection=True;TrustServerCertificate=True;"
```
Comment (pgallik) [inline src/ParcelRegistry.Projections.BackOffice/appsettings.json:6]: should we use this opportunity to use only BackOffice here? Seeing as BackOffice, BackOfficeProjections and BackOfficeProjectionsAdmin are all pointing to the same connectionstring.

File: src/ParcelRegistry.Projections.BackOffice/appsettings.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  },
+
+  "DataDog": {
+    "Enabled": true,
+    "Debug": false,
+    "ServiceName": "parcel-registry-projections-backoffice-dev"
+  },
+
+  "DistributedLock": {
+    "Region": "eu-west-1",
```
Comment (pgallik) [inline src/ParcelRegistry.Projections.BackOffice/appsettings.json:16]: Enabled not required?

**Review by pgallik — CHANGES_REQUESTED** [review]

File: src/ParcelRegistry.Projections.BackOffice/appsettings.json (hunk 2 above)
Comment (ArneD) [reply src/ParcelRegistry.Projections.BackOffice/appsettings.json:16]: default is true, but would be nice to specify it.

File: src/ParcelRegistry.Projections.BackOffice/appsettings.json (hunk 1 above)
Comment (ArneD) [reply src/ParcelRegistry.Projections.BackOffice/appsettings.json:6]: BackOffice and BackOfficeProjections are different, I think they even go to different database.

File: src/ParcelRegistry.Projections.BackOffice/BackOfficeProjections.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            });
+
+            When<Envelope<ParcelAddressWasDetachedBecauseAddressWasRemoved>>(async (_, message, cancellationToken) =>
+            {
+                await using var backOfficeContext = await backOfficeContextFactory.CreateDbContextAsync(cancellationToken);
+                await backOfficeContext.RemoveIdempotentParcelAddressRelation(
+                    new ParcelId(message.Message.ParcelId),
+                    new AddressPersistentLocalId(message.Message.AddressPersistentLocalId),
+                    cancellationToken);
+            });
```
Comment (ArneD) [inline src/ParcelRegistry.Projections.BackOffice/BackOfficeProjections.cs:57]: These shouldn't be done, cause these come via the consumer. Which won't have the "transaction" problem as in the api. But it's also not forbidden to put it here, an extra check :)

File: test/ParcelRegistry.Tests/BackOffice/FakeBackOfficeContext.cs [hunk 4]
```diff
@@ -30,15 +35,32 @@ protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
...
+
+            return base.DisposeAsync();
+        }
     }
 
     public class FakeBackOfficeContextFactory : IDesignTimeDbContextFactory<FakeBackOfficeContext>
     {
+        private readonly bool _dontDispose;
+
+        public FakeBackOfficeContextFactory(bool dontDispose = false)
```
Comment (ArneD) [inline test/ParcelRegistry.Tests/BackOffice/FakeBackOfficeContext.cs:54]: `dispose = true` ?

**Review by ArneD — CHANGES_REQUESTED** [review]

File: src/ParcelRegistry.Projections.BackOffice/appsettings.json (hunk 2 above)
Comment (jvandaal) [reply src/ParcelRegistry.Projections.BackOffice/appsettings.json:16]: Done.

File: test/ParcelRegistry.Tests/BackOffice/FakeBackOfficeContext.cs (hunk 4 above)
Comment (jvandaal) [reply test/ParcelRegistry.Tests/BackOffice/FakeBackOfficeContext.cs]: Changed.

**Review by ArneD — APPROVED** [review]

**Pull request merged**